src/purchase-client.ts

```diff
@@ -0,0 +1,322 @@
+  /**
+   * purchase will start the twitch bits transaction.
+   *
+   * @since 2.4.0
+   *
+   * @throws {TwitchHelperError} Will throw an error if the Twitch Helper didn't load.
+   * 
+   * @param sku the twitch sku of the item to be purchased.
+   *
+   * @example
+   * client.purchase("XXSKU000");
+   */
+  public purchase(sku: string) {
+    if (!window.Twitch?.ext?.bits) {
+      throw new Error('Twitch helper was not loaded.');
+    }
+
+    window.Twitch.ext.bits.useBits(sku);
+  }
+ 
+  /**
+   * Returns a list of Twitch Bits Products for the current channel.
+   * 
+   * @async
+   * @since 2.4.0
+   * 
+   * @throws {TwitchHelperError} Will throw an error if the Twitch Helper didn't load.
+   *
+   * @return {Promise<[]TwitchBitsProduct>} Resolves with an array of {@link TwitchBitsProduct}
+   * objects for each available sku.
+   *
+   * @example
+   * const products = client.getProducts();
+   */
+  public getProducts(): Promise<TwitchBitsProduct[]> {
+    if (!window.Twitch?.ext?.bits) {
+      throw new Error('Twitch helper was not loaded.');
+    }
+    
+    return window.Twitch.ext.bits.getProducts();
+  } 
+
+  /**
+   * onUserPurchase is the interface for adding a post transaction callback.
+   *
+   * @since 2.4.0
+   *
+   * @throws {TwitchHelperError} Will throw an error if the Twitch Helper didn't load.
+   * 
+   * @param callback a function(body)
+   *
+   * @example
+   * client.onUserPurchase(() => {
+   *  console.log("Transaction finished!");
+   * });
+   */
+  public onUserPurchase(callback: (tx: Transaction) => void): void {
+    if (!window.Twitch?.ext?.bits) {
+      throw new Error('Twitch helper was not loaded.');
+    }
+
+    window.Twitch.ext.bits.onTransactionComplete(
+      (tx: TwitchBitsTransaction) => {
+        if (tx.initiator.toLowerCase() === "current_user") {
+          callback(tx);
+        }
+      }
+    );
+  }
+}
+
+// DevPurchaseClient implements the basic 'purchase client' interface.
+// This is used by SDK to provide low-level access to stubbed transactions.
+export class DevPurchaseClient implements PurchaseClient {
+  private purchaseCallbacks: Array<(tx: Transaction) => void> = [];
+
+  /**
+   * purchase will start the Dev purchase transaction.
+   *
+   * @since 2.4.0
+   * 
+   * @param sku the sku of the item to be purchased.
+   *
+   * @example
+   * client.purchase("TESTSKU00");
+   */
+  public purchase(sku: string) {
+    if ("MEDKIT_PURCHASABLE_ITEMS" in window) {
+      const devItems = this.getProducts();
+
+      const products = Object.keys(devItems).map((sku) => devItems[sku]);
+      const foundItem = products.find((sku: string) => sku === sku);
+
+      setTimeout(() => {
+        if (foundItem) {
+          const tx: Transaction = {
+            transactionId: "dev-transaction-id",
+            product: foundItem,
+            userId: "dev-test-user",
+            displayName: "DevTestUser",
+            initiator: "CURRENT_USER",
```

Comment:
The Twitch docs do say that it will always be uppercase. From https://dev.twitch.tv/docs/extensions/reference
```
Valid values:

    CURRENT_USER (the user who executed, or initiated, the transaction).
    OTHER (a user other than CURRENT_USER).
```

But as long as we're lower/uppercasing it and checking that, I guess it doesn't matter.